localstack/services/s3/s3_listener.py

```diff
@@ -323,14 +333,28 @@ def convert_origins_into_list(allowed_origins):
 def append_cors_headers(bucket_name, request_method, request_headers, response):
     bucket_name = normalize_bucket_name(bucket_name)
 
+    # Checking CORS is allowed or not
     cors = BUCKET_CORS.get(bucket_name)
     if not cors:
         return
 
-    origin = request_headers.get('Origin', '')
+    # Cleaning headers
+    for header in CORS_HEADERS:
+        if header in response.headers:
+            del response.headers[header]
+
+    # Fetching origin of the request
+    origin = request_headers.get('Origin')
+
+    if not origin:
+        x_forwarded_header = request_headers.get('X-Forwarded-For').split(', ')
```

Comment:
nit: Better split via regex `re.split(..)` here, to make the logic work independently of whitespaces use (see also comment above).